nodejs/week3/practices/exercise-template/src/backend/api/concerts.js

```diff
@@ -0,0 +1,169 @@
+            const requestedConcert = await concerts.filter(concert=>concert.title.toLowerCase() ===requestTitle);
+           
+    
+            if(titleFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+           
+          }else if(request.query.hasOwnProperty("createdAfter")){
+            
+            const requestDate=new Date(request.query.createdAfter);
+            const concertFinder = concerts.some(concert=> new Date(concert.created_date) > requestDate);
+            const requestedConcert = await concerts.filter(concert => new Date(concert.created_date) > requestDate);
+           
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+   
+          }else if(request.query.hasOwnProperty("band")){
+            const requestBand=request.query.band.toLowerCase();
+            const concertFinder = concerts.some(concert=> concert.band.toLowerCase() ===requestBand);
+            const requestedConcert = await concerts.filter(concert=>concert.band.toLowerCase() ===requestBand);
+           
+    
+            if(concertFinder){
+              response.json(requestedConcert);    
+            }else{ 
+                response.status(404).send('no concert match your request'); 
+            }
+          }else{
+            response.json(concerts);
+          }
+ 
+  } catch (error) {
+    throw error;
+  }
+});
+  
+
+router.post("/", async (request, response) => {
+
+  try {
+     // knex syntax for selecting things. Look up the documentation for knex for further info
+    const concerts = await knex('concerts').insert(request.body);
+   
+    // Alternatively could also insert in the way commented below
+    // but I personally prefer request.body method and using postman.
+   
+    // const concerts = await knex('concerts').insert( 
+    //   {"title": "spring festival",
+    // "band": "many singers",
+    // "venue": "spring celebration",
+    // "created_date": '2021-02-28 15:00:00',
+    // "performance_date": '2021-04-22 17:00:00',
+    // "price": 100});
+   
+    response.json(concerts);
+  } catch (error) {
+    throw error;
+  }
+});
+
+
+router.put("/", async (request,response)=> {
+  try{
+    //api/concerts/{id}	PUT	Updates the concert by id	
+    const concerts = await knex("concerts");
+   
+
+    if(request.query.hasOwnProperty('id')){
+      
+      const concertId=Number(request.query.id);
+      const concertFinder = concerts.some(concert=> concert.id ===concertId);
+      if(concertFinder){
+           const updateConcerts = await knex("concerts")
+           .where({ id: concertId })
+         .update({ band : 'electronic concert' });
+         response.json(`concert with id: ${concertId} was successfully Updated.`)
```

Comment:
great response message, this communicates something to the user. 👍🏾 
Can you make it ``response.send`` instead of json  though?